gd32e10x/cmsis/gd/gd32e10x/include/gd32e10x.h

```diff
@@ -57,7 +57,15 @@ OF SUCH DAMAGE.
   #define HXTAL_VALUE    ((uint32_t)25000000) /*!< value of the external oscillator in Hz */
   #define HXTAL_VALUE_25M  HXTAL_VALUE
   #else
-  #error "Please select the target board type used in your application (in gd32e10x.h file)"
+    /* get external oscillator from KCONFIG */
+    #define HXTAL_VALUE CONFIG_GD32_HXTAL_VALUE
+    #if HXTAL_VALUE == 8000000
+      #define HXTAL_VALUE_8M  HXTAL_VALUE
+    #elif HXTAL_VALUE == 25000000
+      #define HXTAL_VALUE_25M  HXTAL_VALUE
+    #else
+      #error "GD32E10X lib only support 8M and 25M oscillator (HXTAL)"
+    #endif
```

Comment:
There's no need to do that, just drop this commit and do the same as https://github.com/zephyrproject-rtos/zephyr/commit/a93908563631ddba6308587e77cffd70bde30411 (cherry-pick it in your PR while it's not merged, even though it's now approved so it'll go in soon)

Reply:
Oh,I see `HXTAL_VALUE` was defined in cmake script, I'll remove it.
Because of code in https://github.com/zephyrproject-rtos/hal_gigadevice/blob/main/gd32e10x/cmsis/gd/gd32e10x/source/system_gd32e10x.c#L745-L751 ,
I need `HXTAL_VALUE_8M` or `HXTAL_VALUE_25M` marco to make clock config success.